plugins/woocommerce/includes/class-wc-comments.php

```diff
@@ -259,8 +259,8 @@ public static function comment_moderation_recipients( $emails, $comment_id ) {
 	 * @param int $post_id Post ID.
 	 */
 	public static function clear_transients( $post_id ) {
-		if ( 'product' === get_post_type( $post_id ) ) {
-			$product = wc_get_product( $post_id );
+		$product = $post_id ? wc_get_product( $post_id ) : $post_id;
```

Comment:
I think it will be more efficient to keep the check against `get_post_type()`. Otherwise, the other queries for determining the underlying product type will be run before hitting the exception in `WC_Product_Factory::get_product()`

https://github.com/woocommerce/woocommerce/blob/62f844a3f3b63946d4bcd0d369362966e63ff0b3/plugins/woocommerce/includes/class-wc-product-factory.php#L47-L50

```suggestion
		$post_id = absint( $post_id );
		if ( $post_id === 0 || 'product' !== get_post_type( $post_id ) ) {
			return;
		}

		$product = wc_get_product( $post_id );
```